src/main/java/httpserver/httpactions/MethodNotAllowed.java

```diff
@@ -9,9 +9,8 @@
 public class MethodNotAllowed implements Action {
     @Override
     public HttpResponse act() {
-        HttpResponse response = new HttpResponse();
-        response.setResponseLine("405 Method Not Allowed");
-        response.setHeaders(new ArrayList<String>(Arrays.asList("Allow: HEAD, OPTIONS")));
-        return response;
+        return new HttpResponse.HttpResponseBuilder("405 Method Not Allowed")
+                .headers(new ArrayList<String>(Arrays.asList("Allow: HEAD, OPTIONS")))
```

Comment:
This also makes me wonder if our abstraction isn't sufficiently encapsulating some of the underlying details of the HTTP Response. It would be easy to have a typo in the response line. And that typo might lead to an invalid HTTP response. For example

```java
new HttpResponse.HttpResponseBuilder("I am not a valid response line");
```

Something to keep in mind perhaps.

Reply:
Actually we capture a status and not the response line - I renamed to reflect status.

Reply:
Also I separated the status code and status name entries to reduce potential entry errors